instrumentation/github.com/emicklei/go-restful/otelrestful/internal/semconvutil/httpconv_test.go

```diff
@@ -165,47 +166,83 @@ func TestHTTPClientRequestRequired(t *testing.T) {
 }
 
 func TestHTTPServerRequest(t *testing.T) {
-	got := make(chan *http.Request, 1)
-	handler := func(w http.ResponseWriter, r *http.Request) {
-		got <- r
-		w.WriteHeader(http.StatusOK)
-	}
+	testFn := func(requestModifierFn func(r *http.Request), opts HTTPServerRequestOptions, expectClientIP string) func(t *testing.T) {
```

Comment:
This fn could get a lot of parameter if we start adding more specific attributes.
How about following this pattern instead for subtests? https://go.dev/blog/subtests

The struct would have attributes such as the HTTP request, and the options.
And a `wantAttributes` attribute.

The test would do the same thing as before, except the expected attributes would be a dynamic value.

Reply:
Changed